### ISSUE TITLE
Outline view does not work for Zig

### ISSUE THREAD

**Issue by StringNick** [zed-industries/zed#7357]
### Check for existing issues

- [X] Completed

### Describe the feature

Support symbol picker for a file, currently symbol picker working by workspace.

**Comment by ConradIrwin** [comment]
Hey StringNick! I think the feature you're looking for is "outline view" `cmd-shift-o` by default?

Feel free to re-open if not.

**Comment by StringNick** [comment]
yes this is correct, but its not working for ziglang, should i open another one issue, or we can change this one?

**Comment by StringNick** [comment]
and inlay hint for zig not working fully, how its in zed:
<img width="487" alt="image" src="https://github.com/zed-industries/zed/assets/13052752/479dd7fb-64bd-4c3d-8087-730bf22344a3">
how this one in vscode:
<img width="517" alt="image" src="https://github.com/zed-industries/zed/assets/13052752/734bf260-48e3-4b05-a3fb-1efc967d8223">
settings in zed: 
```  "inlay_hints": {
    "enabled": true,
    "show_type_hints": true,
    "show_parameter_hints": true,
    "show_other_hints": true
  }```

**Comment by ConradIrwin** [comment]
Re-opened and updated.

The fix will be to create a file `crates/zed/src/languages/zig/outline.scm` that contains the tree-sitter queries to construct the outline.

The easiest way to do this is to open a zig file, and then use `cmd-shift-p "debug: open syntax tree view"` to see the tree, and copy another language for inspiration.

If you'd like to pair on this feel free to book time with me: https://calendly.com/conradirwin/pairing

**Comment by StringNick** [comment]
@ConradIrwin but right now zed support symbol picker through tree sitter queries, but as i know in lsp specification they have symbol picker, maybe better to use LSP outline (also workspace outline is exist), than writing for everylanguage tree sitter query?

**Comment by JosephTLyons** [comment]
This is available in Zed [v0.126.0-pre](https://zed.dev/releases/preview/0.126.0).
